src/scrabble.js

```diff
@@ -1,5 +1,116 @@
-function scrabble() {
+// Input word here (moved to top for ease of access)
+const scrabbleInput = 'P[i]nt'
 
+const pointsMap = {
+  A: 1,
+  E: 1,
+  I: 1,
+  O: 1,
+  U: 1,
+  L: 1,
+  N: 1,
+  R: 1,
+  S: 1,
+  T: 1,
+  D: 2,
+  G: 2,
+  B: 3,
+  C: 3,
+  M: 3,
+  P: 3,
+  F: 4,
+  H: 4,
+  V: 4,
+  W: 4,
+  Y: 4,
+  K: 5,
+  J: 8,
+  X: 8,
+  Q: 10,
+  Z: 10
+}
+
+scrabble(scrabbleInput)
+
+function scrabble(word) {
+  // Guard condition for null
+  if (word === null) {
+    console.log(`Why would you enter null? This gives you 0 points.`)
+    return 0
+  }
+
+  // If it's not null we move on
+  // Setting up initial points tally at 0
+  let points = 0
+
+  // Coverting the given word to upper case
+  const wordUpperCase = word.toUpperCase()
+
+  // Splitting the given word letter by letter
+  const wordSplit = wordUpperCase.split('')
+
+  // Displaying the letters to the user
+  console.log(`Your letters are the following: ${wordSplit}`)
+
+  // Awarding points for each letter
+  for (const character of wordSplit) {
+    points += letterToPoints(character)
+  }
+
+  // Checking for double and triple letters
+  points = doubleAndTripleCharacterChecker(wordSplit, points)
+
+  // Checking for double and triple words
+  points = doubleAndTripleWordChecker(wordSplit, points)
+
+  // Logging the final points tally to the console
+  console.log(`This gives you ${points} points in Scrabble!`)
+
+  // Returning the final points tally
```

Comment:
Separation of each process with their own comments makes the file really readable, if I were to nit-pick I'd say maybe too much? But that might just be a personal preference.

Reply:
Yeah I may have gone overboard...